pkg/state/impl/inmem/collection.go

```diff
@@ -106,51 +107,66 @@ func (collection *ResourceCollection) inject(resource resource.Resource) {
 }
 
 // Create a resource.
-func (collection *ResourceCollection) Create(ctx context.Context, resource resource.Resource, owner string) error {
-	resource = resource.DeepCopy()
+func (collection *ResourceCollection) Create(ctx context.Context, res resource.Resource, owner string) error {
+	resCopy := res.DeepCopy()
 
-	if err := resource.Metadata().SetOwner(owner); err != nil {
+	if err := resCopy.Metadata().SetOwner(owner); err != nil {
 		return err
 	}
 
 	collection.mu.Lock()
 	defer collection.mu.Unlock()
 
-	if _, exists := collection.storage[resource.Metadata().ID()]; exists {
-		return ErrAlreadyExists(resource.Metadata())
+	if _, exists := collection.storage[resCopy.Metadata().ID()]; exists {
+		return ErrAlreadyExists(resCopy.Metadata())
 	}
 
+	version, err := resource.ParseVersion("1")
+	if err != nil {
+		return err
+	}
+
+	updated := time.Now()
+
+	resCopy.Metadata().SetVersion(version)
+	resCopy.Metadata().SetUpdated(updated)
```

Comment:
we can probably skip setting `Updated` here, as `NewMetadata` automatically sets `created = updated = now()`

Reply:
done